site/src/pages/api/rewrites/sandboxed-block-demo.api.ts

```diff
@@ -0,0 +1,133 @@
+import { JSONObject } from "blockprotocol";
+import { NextApiHandler } from "next";
+
+import { readBlockDataFromDisk, readBlocksFromDisk } from "../../../lib/blocks";
+
+/**
+ * @todo remove after fixing
+ * - https://blockprotocol.org/@jmackie/blocks/quote
+ * - https://blockprotocol.org/@shinypb/blocks/emoji-trading-cards
+ * - https://blockprotocol.org/@kickstartds/blocks/button
+ */
+const hotfixExternals = (externals: JSONObject | undefined): JSONObject => {
+  return (
+    externals ?? {
+      react: "17.0.2",
+      lodash: "4.17.21",
+      twind: "0.16.16",
+    }
+  );
+};
+
+/**
+ * @todo potentially remove after building blocks to ESM
+ */
+const hotfixPackageName = (packageName: string): string => {
+  return packageName === "lodash" ? "lodash-es" : packageName;
+};
+
+const handler: NextApiHandler = async (req, res) => {
+  if (req.url?.startsWith("/api/rewrites/")) {
+    res.status(404).send("Not found");
+    return;
+  }
+
+  const catalog = readBlocksFromDisk();
+
+  const packagePath = `${req.query.shortname}/${req.query.blockslug}`;
+
+  const blockMetadata = catalog.find(
+    (metadata) => metadata.packagePath === packagePath,
+  );
+
+  if (!blockMetadata) {
+    res.status(404);
+    res.write("Block not found");
+    res.end();
+    return;
+  }
+
+  const { source } = await readBlockDataFromDisk(blockMetadata);
+  const reactVersion = blockMetadata.externals?.react ?? "17.0.2";
+  const mockBlockDockVersion = "0.0.9";
```

Comment:
This is going to be tedious to maintain, is there a way we can get it automatically?